rust_snuba/src/processors/generic_metrics.rs

```diff
@@ -78,6 +81,60 @@ enum MetricValue {
     },
 }
 
+#[derive(Debug, Deserialize)]
+#[serde(tag = "format", rename_all = "lowercase")]
+enum EncodedSeries<T> {
+    Array { data: Vec<T> },
+}
+
+impl<T> EncodedSeries<T> {
+    fn into_vec(self) -> Vec<T> {
+        match self {
+            EncodedSeries::Array { data } => data,
+        }
+    }
+}
+
+fn encoded_series_deserializer<'de, T, D>(deserializer: D) -> Result<EncodedSeries<T>, D::Error>
+where
+    T: Deserialize<'de>,
+    D: Deserializer<'de>,
+{
+    struct Visitor<U> {
```

Comment:
nit:

```suggestion
    struct Visitor<U>(PhantomData<U>);
```

Usually define these Visitors as a unit struct.

Reply:
Fixed, thank you